### DISCUSSION TITLE
[v3] Value of input inside $slot gets assigned to Alpine property

### DISCUSSION THREAD

**Discussion by klaasgeldof** [livewire/livewire#6486] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.0-beta.11

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I have this Blade component `modal`:

```blade
<div x-show="show" x-cloak x-data="{ show: @entangle($attributes->wire('model')) }">
    {{ $slot }}
</div>
```

And can use it as follows in a Livewire component:

```blade
<x-modal wire:model.live="openModal">
    <input type="text" />
</x-modal>
```

Everything works fine. However, as soon as I add `{{ $attributes->class('...') }}` before `x-data`:

```blade
<div x-show="show" x-cloak {{ $attributes->class('...') }} x-data="{ show: @entangle($attributes->wire('model')) }">
    {{ $slot }}
</div>
```

then whatever I type in the text input becomes the value of `show`.

![CleanShot 2023-08-24 at 11 30 23](https://github.com/livewire/livewire/assets/35819/b1f1819d-03b3-44f0-831e-bf1c86593c16)

When I change `{{ $attributes->class('...') }}` to a regular `class="..."` everything works fine again...

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by N1ebieski** [comment, marked as answer]
Try:

```php
$attributes->class('...')->filter(fn ($value, $key) => !Str::startsWith($key, 'wire:'))
```

**Reply by klaasgeldof** [reply]
Oh yeah, that did the trick! Didn't realize the extra `wire:model` on the modal div caused the problem.

Just a small typo in your code: `starts_with` should be `startsWith`

Thanks!!

**Reply by joshhanley** [reply]
@N1ebieski and @klaasgeldof you can make it simpler by using `$attributes->class('...')->whereDoesntStartWith('wire:');` see https://laravel.com/docs/10.x/blade#filtering-attributes
